packages/cloud_firestore/cloud_firestore_platform_interface/lib/src/method_channel_cloud_firestore.dart

```diff
@@ -0,0 +1,356 @@
+  void _handleQuerySnapshot(MethodCall call) {
+    final int handle = call.arguments['handle'];
+    _queryObservers[handle].add(call.arguments);
+  }
+
+  //TODO: Type this return
+  @override
+  Future<Map<dynamic, dynamic>> getQueryDocuments(String app, {
+    @required String path,
+    bool isCollectionGroup,
+    Map<String, dynamic> parameters,
+    String source,
+  }) => channel.invokeMapMethod<String, dynamic>(
+      'Query#getDocuments',
+      <String, dynamic>{
+        'app': app,
+        'path': path,
+        'isCollectionGroup': isCollectionGroup,
+        'parameters': parameters,
+        'source': source,
+      },
+    );
+
+  // Transaction
+  // TODO: Type this return
+  @override
+  Future<Map<String, dynamic>> getTransaction(String app, {
+    @required String path,
+    @required int transactionId,
+  }) => channel
+        .invokeMapMethod<String, dynamic>('Transaction#get', <String, dynamic>{
+      'app': app,
+      'transactionId': transactionId,
+      'path': path,
+    });
+
+  @override
+  Future<void> deleteTransaction(String app, {
+    @required String path,
+    @required int transactionId,
+  }) => channel
+        .invokeMethod<void>('Transaction#delete', <String, dynamic>{
+      'app': app,
+      'transactionId': transactionId,
+      'path': path,
+    });
+
+  @override
+  Future<void> updateTransaction(String app, {
+    @required String path,
+    @required int transactionId,
+    Map<String, dynamic> data,
+  }) => channel
+        .invokeMethod<void>('Transaction#update', <String, dynamic>{
+      'app': app,
+      'transactionId': transactionId,
+      'path': path,
+      'data': data,
+    });
+
+  @override
+  Future<void> setTransaction(String app, {
+    @required String path,
+    @required int transactionId,
+    Map<String, dynamic> data,
+  }) => channel
+        .invokeMethod<void>('Transaction#set', <String, dynamic>{
+      'app': app,
+      'transactionId': transactionId,
+      'path': path,
+      'data': data,
+    });
+
+  // Write Batch
+  @override
+  Future<dynamic> createWriteBatch(String app) => channel.invokeMethod<dynamic>(
+            'WriteBatch#create', <String, dynamic>{'app': app});
+
+  @override
+  Future<void> commitWriteBatch({
+    @required dynamic handle,
+  }) => channel.invokeMethod<void>(
+          'WriteBatch#commit', <String, dynamic>{'handle': handle});
+
+  @override
+  Future<void> deleteWriteBatch(String app, {
+    @required dynamic handle,
+    @required String path,
+  }) => channel.invokeMethod<void>(
+            'WriteBatch#delete',
+            <String, dynamic>{
+              'app': app,
+              'handle': handle,
+              'path': path,
+            },
+          );
+
+  @override
+  Future<void> setWriteBatchData(String app, {
+    @required dynamic handle,
+    @required String path,
+    Map<String, dynamic> data,
+    Map<String, dynamic> options,
+  }) => channel.invokeMethod<void>(
```

Comment:
Only use `=>` when everything, including the function declaration, fits on a single line.